src/ripple/app/consensus/RCLValidations.cpp

```diff
@@ -227,4 +227,26 @@ handleNewValidation(
     return shouldRelay;
 }
 
+void
+filterValsWithnUnl(
+    std::vector<std::shared_ptr<STValidation>>& validations,
+    hash_set<NodeID> const& nUnl)
+{
+    if (nUnl.empty())
```

Comment:
This entire function basically reduces to this (untested, typed on my phone):

    /* Remove validations that are from validators on the negative UNL. */
    if (!nUnl.empty())
    {
        validations.erase(
            std::remove_if(
                validations.begin(), validations.end(), [&nUnl](auto const& v) {
                    return nUnl.find(v->getNoideID()) != nUnl.end();
                }, validations.end());
    }

We call this function from several places, which is unfortunate. If it was called once, we could simply inline it with no loss in readability.

Reply:
Thanks, changed